### ISSUE TITLE
Using forward hook for auxiliary loss

### ISSUE THREAD

**Issue by 1ho0jin1** [yoshitomo-matsubara/torchdistill#152]
Hello!

I was trying to find a way to use the intermediate features of a pretrained ResNet without changing the architecture, then found this github repo by the 2020 RPRR paper _torchdistill: A Modular, Configuration-Driven Framework for Knowledge Distillation_.

In the paper it mentions:
_Taking an advantage of forward hook paradigm in PyTorch [30], torchdistill supports introducing such auxiliary modules without altering the original implementations
of the models._

I was very excited to find this repo since this is exactly what I was looking for!

However, it is difficult for me to actually spot an example code for this.

Can anyone generously highlight where in this repo I can find a code regarding this, or provide an example for me?

Thanks for the help in advance :)

**Comment by yoshitomo-matsubara** [comment]
Hi @1ho0jin1 

Thank you for your interest in torchdistill!
I think that [this notebook](https://github.com/yoshitomo-matsubara/torchdistill/blob/main/demo/extract_intermediate_representations.ipynb) would be a perfect example for you.

The above example is independent from generalized pipelines in torchdistill, so you can integrate it to your own code by installing torchdistill (pip install torchdistill). If you'd like to use ResNet with the generalized pipeline e.g., for ImageNet or CIFAR datasets, just let me know.

**Comment by 1ho0jin1** [comment]
@yoshitomo-matsubara Thanks for such a quick reply.

This was exactly what I was looking for!

And thank you again for sharing this great work :+1: 

I'll close this issue now :)
